### PR TITLE
Azure bulk loading - filters results using tags

### PR DESCRIPTION
**Pull request by usmansaleem**
## PR Description
Azure bulk loading - filter results based on tags. Use cli option to specify the tag names and values pair. Added acceptance test to return results based on tag.

```
      --azure-secrets-tags=<TAG_NAME=TAG_VALUE>
                            Specify optional tags to filter the secrets. For
                              example --azure-secrets-tags ENV=prod
                              --azure-secrets-tags=CLIENT=A.
```
## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

File: signing/src/main/java/tech/pegasys/web3signer/signing/config/AzureKeyVaultParameters.java [hunk 1]
```diff
@@ -25,4 +28,8 @@ public interface AzureKeyVaultParameters {
   String getClientId();
 
   String getClientSecret();
+
+  default Map<String, String> getTags() {
```
Comment (jframe) [inline signing/src/main/java/tech/pegasys/web3signer/signing/config/AzureKeyVaultParameters.java:32]: This default method is only used in one place the AzureSecretSigningMetadata. Would be clearer to just return the emptyMap for that one case.

File: signing/src/main/java/tech/pegasys/web3signer/signing/config/AzureKeyVaultParameters.java [hunk 2]
```diff
@@ -12,6 +12,9 @@
  */
 package tech.pegasys.web3signer.signing.config;
 
+import java.util.Collections;
+import java.util.Map;
+
 public interface AzureKeyVaultParameters {
```
Comment (jframe) [inline signing/src/main/java/tech/pegasys/web3signer/signing/config/AzureKeyVaultParameters.java:18]: The AwsKeySigningMetadata doesn't have an implementation so we won't be able to use tags in the key config files

File: signing/src/main/java/tech/pegasys/web3signer/signing/config/AzureKeyVaultParameters.java (hunk 1 above)
Comment (usmansaleem) [reply signing/src/main/java/tech/pegasys/web3signer/signing/config/AzureKeyVaultParameters.java:32]: agreed, will make change.

File: signing/src/main/java/tech/pegasys/web3signer/signing/config/AzureKeyVaultParameters.java (hunk 2 above)
Comment (usmansaleem) [reply signing/src/main/java/tech/pegasys/web3signer/signing/config/AzureKeyVaultParameters.java:18]: tags only make sense for bulk load. For key config file, we always specify the "secret name" to load.

File: commandline/src/main/java/tech/pegasys/web3signer/commandline/PicoCliAzureKeyVaultParameters.java [hunk 3]
```diff
@@ -61,6 +64,14 @@ public class PicoCliAzureKeyVaultParameters implements AzureKeyVaultParameters {
       paramLabel = "<CLIENT_SECRET>")
   private String clientSecret;
 
+  @Option(
+      names = {"--azure-secrets-tags"},
+      description =
+          "Specify optional tags to filter the secrets. "
+              + "For example --azure-secrets-tags ENV=prod --azure-secrets-tags=CLIENT=A.",
+      paramLabel = "<TAG_NAME=TAG_VALUE>")
```
Comment (siladu) [inline commandline/src/main/java/tech/pegasys/web3signer/commandline/PicoCliAzureKeyVaultParameters.java:72]: AWS implements --aws-secrets-tag-names-filter and --aws-secrets-tag-values-filter
so this is inconsistent from a UX perspective, though not sure how much that matters in this instance.

Is there a reason this was implemented differently?

**Review by jframe — APPROVED** [review]

File: signing/src/main/java/tech/pegasys/web3signer/signing/config/metadata/AzureSecretSigningMetadata.java [hunk 4]
```diff
@@ -87,4 +90,11 @@ public String getSecretName() {
   public ArtifactSigner createSigner(final ArtifactSignerFactory factory) {
     return factory.create(this);
   }
+
+  @Override
+  public Map<String, String> getTags() {
+    // tags support is not applicable for config file mode as
+    // user is already providing the secret name to load the secret from.
+    return Collections.emptyMap();
```
Comment (siladu) [inline signing/src/main/java/tech/pegasys/web3signer/signing/config/metadata/AzureSecretSigningMetadata.java:98]: Could/should we throw unimplemented exception here? Feels like we should log a warning too.

**Pull request merged**